src/libsync/theme.cpp

```diff
@@ -477,33 +477,41 @@ bool Theme::aboutShowCopyright() const
 #ifndef TOKEN_AUTH_ONLY
 QIcon Theme::syncStateIcon(SyncResult::Status status, bool sysTray, bool sysTrayMenuVisible) const
 {
-    // FIXME: Mind the size!
+    return syncStateIcon(SyncResult { status }, sysTray, sysTrayMenuVisible);
+}
+QIcon Theme::syncStateIcon(const SyncResult &status, bool sysTray, bool sysTrayMenuVisible) const
+{
     QString statusIcon;
 
-    switch (status) {
-    case SyncResult::Undefined:
-        // this can happen if no sync connections are configured.
-        statusIcon = QStringLiteral("state-information");
-        break;
+    switch (status.status()) {
     case SyncResult::NotYetStarted:
+        Q_FALLTHROUGH();
     case SyncResult::SyncRunning:
         statusIcon = QStringLiteral("state-sync");
         break;
     case SyncResult::SyncAbortRequested:
+        Q_FALLTHROUGH();
     case SyncResult::Paused:
         statusIcon = QStringLiteral("state-pause");
         break;
     case SyncResult::SyncPrepare:
+        Q_FALLTHROUGH();
     case SyncResult::Success:
+        if (status.hasUnresolvedConflicts()) {
+            return syncStateIcon(SyncResult { SyncResult::Problem }, sysTray, sysTrayMenuVisible);
+        }
```

Comment:
We need to do something about the formatter here. The placing of these braces is misleading. I needed to read it a second time to see that it's constructing a new object.